### PR TITLE
r/aws_elasticsearch_domain: export kibana endpoint

### LINKED ISSUE #2802: Kibana Attribute Not available for elasticsearch
I want to be able to reference the Kibana endpoint for DNS or anything else I would like use it for later.  Currently there is only id, access_policies and domain_name that are attributes that are exported. 

Can we add kibana endpoint? 
$ terraform -version
Terraform v0.11.1
+ provider.aws v1.6.0

```
  id = esd-policy-dev-logs
  access_policies = {"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":{"AWS":"*"},"Action":"es:*","Resource":"arn:aws:es:us-east-********/*"}]}
  domain_name = dev-logs
```

### PR DESCRIPTION
**Pull request by loivis**
Fixes #2802 

@fewknow, is this what you want, or rather export with `https://` scheme? The `endpoint` for elasticsearch domain itself is just a dns record, without any scheme.

### CODE REVIEW

**Comment by fewknow** [comment]
@loivis looks like this will satisfy my use case.   I just wanted to be able to assign DNS to this in terraform so my end users can hit kibana.dev.corvestacloud.net.   

Thanks

File: aws/resource_aws_elasticsearch_domain.go [hunk 1]
```diff
@@ -480,12 +484,14 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 		if err != nil {
 			return err
 		}
+		d.Set("kibana_endpoint", endpoints["vpc"].(string)+"/_plugin/kibana/")
```
Comment (Ninir) [inline aws/resource_aws_elasticsearch_domain.go]: Could we make use of a function that build this endpoint for us here and below?

**Comment by loivis** [comment]
Hi @Ninir 

I added documentation for the attribute and extracted the function to populate kibana endpoint.

Regarding test, I just add simple check in acceptance test. Not sure if we need this check when setting the attribute. It is only basic string concatenation of `endpoint` attribute and `/_plugin/kibana/`

**Comment by loivis** [comment]
```
⎇  make testacc TEST=./aws TESTARGS='-run=TestAccAWSElasticSearchDomain_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSElasticSearchDomain_basic -timeout 120m
=== RUN   TestAccAWSElasticSearchDomain_basic
--- PASS: TestAccAWSElasticSearchDomain_basic (870.58s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	870.636s
```

File: website/docs/r/elasticsearch_domain.html.markdown [hunk 2]
```diff
@@ -109,6 +109,7 @@ The following attributes are exported:
 * `arn` - Amazon Resource Name (ARN) of the domain.
 * `domain_id` - Unique identifier for the domain.
 * `endpoint` - Domain-specific endpoint used to submit index, search, and data upload requests.
+* `kibana_endpoint` - Domain-specific endpoint for kibana without https scheme, the same as `endpoint`.
```
Comment (Ninir) [inline website/docs/r/elasticsearch_domain.html.markdown]: By "the same as `endpoint`", do you mean the same as in "without https"?
If so, I think this is misleading with the beginning of the sentence, what do you think? :)

Comment (loivis) [reply website/docs/r/elasticsearch_domain.html.markdown]: Exactly. Now removed those confusing words.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**